barretenberg/cpp/src/barretenberg/vm2/simulation/sha256.cpp

```diff
@@ -0,0 +1,51 @@
+#include "barretenberg/vm2/simulation/sha256.hpp"
+
+#include <cstdint>
+#include <memory>
+#include <sys/types.h>
+
+#include "barretenberg/vm2/simulation/lib/sha256_compression.hpp"
+
+namespace bb::avm2::simulation {
+
+void Sha256::compression(ContextInterface& context,
```

Comment:
You reminded me that this probbaly needs to return sth

Reply:
hmm maybe but with the current signature it should not. i think it would if it has "pure" inputs 

`array<u32, 8> Sha256::compression(ContextInterface& context,MemoryAddress state_addr, array<u32, 8> input)`

so only the state_addr is passed in as an offset